### PR TITLE
Navigation: Fix 'ariaLabel' block support

### LINKED ISSUE #66935: Add the ability to manually define an aria-label for the navigation block
The core navigation block uses the name of the Navigation post that is assigned to it as it's aria label. 

This behavior is not documented/called out anywhere. 

Since the name of a navigation menu in the admin isn't necessarily very prescriptive to a user using a screen reader on the front end of the site, it would be great to be able to decouple this aria-label from the title of the navigation post.

We should still leave that behavior as the default/fallback in case no label gets specified manually. However, having the ability to define a custom label via the advanced settings panel would be a great improvement.

This way that actual label could already get defined in code rather than relying on the user to set it correctly.

### PR DESCRIPTION
**Pull request by up1512001**
## What?
Added text control field to navigation block to specify other accessible name apart from menu name.

## Why?
Closes https://github.com/WordPress/gutenberg/issues/66935

## How?
- created `accessibleMenuName` attribute and storing name provided in text field.
- setting `aria-lable` to with provided attribute and if that's empty then applying fallback label with menu name.

## Testing Instructions
1. open post editor & add navigation block
2. go to `Settings > Advance` and add any text to `Accessible menu name` field.
3. check front end navigation block aria-label will be updated with new provided text.

## Screenshots or screencast 

https://github.com/user-attachments/assets/8f20b10b-9dba-48d2-8242-d1a99038771e

### CODE REVIEW

File: packages/block-library/src/navigation/block.json [hunk 1]
```diff
@@ -84,6 +84,10 @@
 		"templateLock": {
 			"type": [ "string", "boolean" ],
 			"enum": [ "all", "insert", "contentOnly", false ]
+		},
+		"accessibleMenuName": {
```
Comment (fabiankaegy) [inline packages/block-library/src/navigation/block.json]: Hey 👋 

Based on https://github.com/WordPress/gutenberg/pull/41744 I would think we actually want to use the `ariaLabel` name for the attribute 🤔

Comment (up1512001) [reply packages/block-library/src/navigation/block.json]: @fabiankaegy rename `accessibleMenuName` to `ariaLabel`.

**Review by fabiankaegy — COMMENTED** [review]
Thank you so much for your quick PR here :) This is awesome! 

From my perspective this looks great and tests week! 

Since it was my ticket though, I want to leave a bit of time for others like @getdave to chime in and provide their feedback :)

**Comment by fabiankaegy** [comment]
Actually @up1512001 looks like the test failure here in CI is legit. Seems like some of the test snapshots need to be updated

**Review by getdave — COMMENTED** [review]
Based on the rationale provided by @fabiankaegy in the Issue this seems logical to me. Do we have a precedent in any other Core blocks?

**Comment by fabiankaegy** [comment]
@getdave As of https://github.com/WordPress/gutenberg/pull/41744 we have a block support for `aria-label` that gets used in a few places. It uses the `ariaLabel` attribute under the hood. It does not provide a UI for it however. So I'd be okay with changing this PR to only add the attribute but not add a UI just yet if we are concerned about adding it. 

Maybe @richtabor has an opinion as I know he has spent some time trying to clean up the Advanced panel

**Comment by Mamaduka** [comment]
I agree. Let's solve this using the method from #41744.

**Comment by getdave** [comment]
Let's be consistent with the other approaches then. I'm curious as to why we didn't provide a UI in the Group block.

**Comment by fabiankaegy** [comment]
@getdave 
This is what the other issue said:
> This implementation does NOT add a UI for the aria-label. A visible option would have the potential to confuse users, and could cause more harm than good if they enter the wrong thing or misunderstand the purpose.
However, allowing users & theme-authors to manually define an aria-label in their templates is something that can significantly enhance the a11y of themes, without exposing any option to users. Furthermore, implementing this as a hook will allow us to implement a UI if at some point in the future we choose to do so.

And agreed 👍  @up1512001 can you make that update? :)

**Comment by getdave** [comment]
> @getdave This is what the other issue said:

Thanks. I should have looked at that more carefully. That seems entirely logical. We can always add a control later _if_ we need.

**Comment by t-hamano** [comment]
I'm a bit confused, since the Navigation block [_already_ supports `ariaLabel` as a block support](https://github.com/WordPress/gutenberg/blob/ea3a85b0daf5b28bdbdb975cdd3582ba6b0f01c0/packages/block-library/src/navigation/block.json#L107), developer should be able to override `aria-label` via the code editor etc, right?

**Comment by up1512001** [comment]
> I'm a bit confused, since the Navigation block [_already_ supports `ariaLabel` as a block support](https://github.com/WordPress/gutenberg/blob/ea3a85b0daf5b28bdbdb975cdd3582ba6b0f01c0/packages/block-library/src/navigation/block.json#L107), developer should be able to override `aria-label` via the code editor etc, right?

This is dynamic block and inside editor user only sees like 
```
<!-- wp:navigation {"ref":51} /-->
```

that's why `ariaLabel` help to provide any meaningful explanation to menu like this 
```
<!-- wp:navigation {"ref":51,"ariaLabel":"global header"} /-->
```

Let me know if I am missing anything 🙇‍♂️

**Comment by t-hamano** [comment]
So does that mean that the current Navigation block `ariaLabel` block support doesn't work unless we explicitly add the `ariaLabel` attribute? Looking at #54418, it seems to work as expected with just the `ariaLabel` block support.

Or is this PR trying to change how aria-label attributes are handled server-side?

Sorry if I misunderstood something. I'm trying to understand what this PR is trying to achieve.

**Comment by Mamaduka** [comment]
You're correct, @t-hamano! The `aria-label` support was introduced in #54418, but it stopped working properly after that.

To get it working again, I think we just need to fix the logic in the `WP_Navigation_Block_Renderer::get_navigation_name` method.

**Comment by up1512001** [comment]
@Mamaduka added early return in `WP_Navigation_Block_Renderer::get_navigation_name` if `ariaLable` is not empty.

https://github.com/user-attachments/assets/5cb104be-ff56-43e3-97c1-3f816bde340c

**Review by Mamaduka — APPROVED** [review]
Thanks for the follow-ups, @up1512001!

Everything is working as expected ✅

**Pull request merged**